packages/evm/contracts/Yaho.sol

```diff
@@ -38,14 +38,15 @@ contract Yaho is IMessageDispatcher, MessageHashCalculator {
     function relayMessagesToAdapters(
         uint256[] memory messageIds,
         address[] memory adapters,
-        address[] memory destinationAdapters
+        address[] memory destinationAdapters,
+        uint256 gas
```

Comment:
I wonder if this should either be an array of gas values or if destination adapters should be a struct like this?
```solidity
struct DestinationAdapter {
    address adapter,
    uint256 gas
}
```

The reason being that different adapters may or may not need this value, and of those that do, they may use use it for very different things.